### ISSUE TITLE
Goob loses on time at 01s10s in CuteChess

### ISSUE THREAD

**Issue by tissatussa** [Gabrielkaos/ChessEngineGoobC#4]
in CuteChess v1.2.0 i test with time control 01s10s : 1 second per move with 10 seconds bonus - i think this way the engines play a fair time per move, they can even gather time by making some instant moves.

suddenly Goob v1.8.9 time management failed :

![Goob-White-loses-on-time](https://github.com/Gabrielkaos/ChessEngineGoobC/assets/1109281/7f96fd73-2a4f-46bf-9955-807085bac849)

-with 'normal' tc all Goob runs OK

[ i'm on Xubuntu 22.04 ]

**Comment by Gabrielkaos** [comment]
> in CuteChess v1.2.0 i test with time control 01s10s : 1 second per move with 10 seconds bonus - i think this way the engines play a fair time per move, they can even gather time by making some instant moves.

yeah havent really integrated a good time man in goob, currently uses the most basic, thanks for pointing out a weakness in the engine, I'll improve the time man in the next update, big thanks :)

**Comment by tissatussa** [comment]
> ..I'll improve the time man in the next update..

lately i wrote another Issue at https://github.com/RedBedHed/Homura/issues/4 which is a rather simple engine and lacks proper Time Control (tc) also .. i descibe some common method (the 30 move one), maybe you can use it .. however, it seems your tc code does not properly include the bonus time (10 sec) AND my very low starting time (1 sec) "into thinking periods".
btw. in CuteChess the bonus time is added BEFORE an engine starts thinking.

**Comment by Gabrielkaos** [comment]
> lately i wrote another Issue at [RedBedHed/Homura#4](https://github.com/RedBedHed/Homura/issues/4) which is a rather simple engine and lacks proper Time Control (tc) also .. i descibe some common method (the 30 move one), maybe you can use it .. however, it seems your tc code does not properly include the bonus time (10 sec) AND my very low starting time (1 sec) "into thinking periods". btw. in CuteChess the bonus time is added BEFORE an engine starts thinking.

Thanks I'll look into them ;), I'll update goob soon
